addons/sale/models/account_move_line.py

```diff
@@ -45,15 +48,32 @@ def _prepare_analytic_line(self):
 
         return values_list
 
+    def _update_analytic_line_sol(self, vals):
```

Comment:
name doesn't seem too descriptive to me

can't we do this method in `_sale_create_reinvoice_sale_line` , like:

```diff
  # no reinvoice as no sales order was found
  if not sale_order:                                
     continue

+ # no reinvoice as it is a refund
+ if move_line.move_id.move_type in ('out_refund', 'in_refund'):
+     sale_order_line = sale_order.order_line.filtered(lambda l: l.qty_delivered_method == 'analytic' and l.product_id == move_line.product_id)
+     map_move_sale_line[move_line.id] = sale_order_line.id
```
instead of the change in _prepare_analytic_line and the new _update_analytic_line_sol

it seems like there is already some case were we keep the existing sale line anyway:

```py
# find the existing sale.line or keep its creation values to process this in batch
sale_line = None
if move_line.product_id.expense_policy == 'sales_price' and move_line.product_id.invoice_policy == 'delivery':  # for those case only, we can try to reuse one
```